react/src/layout/report.js

```diff
@@ -0,0 +1,297 @@
+        let simulationInfoPromise = useContext(ContextSimulationInfoPromise);
+        let appName = useContext(ContextAppName);
+
+        let [reportData, updateReportData] = useState(undefined);
+
+        let reportEventsVersionRef = useRef(uuidv4())
+
+        let shown = true;
+
+        if(shownConfig) {
+            shown = useEvaluatedInterpString(modelsWrapper, shownConfig);
+        }
+
+        let frameIdDependencies = frameIdFields.map(f => new Dependency(f));
+
+        let frameIdDependencyGroup = new HookedDependencyGroup({ 
+            dependencies: frameIdDependencies,
+            modelsWrapper,
+            schemaModels: schema.models
+        })
+
+        let hookedFrameIdDependencies = frameIdDependencies.map(frameIdDependencyGroup.getHookedDependency);
+
+        useEffect(() => {
+            let reportEventsVersion = uuidv4();
+            reportEventsVersionRef.current = reportEventsVersion;
+            
+            reportEventManager.onReportData(reportGroupName, (simulationData) => {
+                if(reportEventsVersionRef.current !== reportEventsVersion) {
+                    return; // guard against concurrency with older versions
+                }
+
+                updateReportData(undefined);
+
+                let { state } = simulationData;
+                if(state == "completed") {
+                    simulationInfoPromise.then(({simulationId}) => {
+                        let { frameCount, computeJobHash, computeJobSerial } = simulationData;
+                        let frameId = getFrameId({
+                            frameIndex: frameCount - 1, // pick last frame for now
+                            reportName,
+                            simulationId,
+                            computeJobHash,
+                            computeJobSerial,
+                            appName,
+                            hookedDependendies: hookedFrameIdDependencies
+                        })
+
+                        getSimulationFrame(frameId).then(simulationData => {
+                            updateReportData(simulationData);
+                        });
+                    })
+                }
+            })
+        })
+
+        let layoutElement = this.layoutsWrapper.getLayoutForConfig(reportLayout);
+        let panelController = useContext(ContextPanelController);
+
+        useEffect(() => {
+            panelController.setShown(shown && !!reportData);
+        }, [shown, !!reportData])
+
+        let VisualComponent = reportData ? layoutElement.component : undefined;
+
+        // set the key as the key for the latest request sent to make a brand new report component for each new request data
+        return (
+            <>
+                {VisualComponent && shown && <VisualComponent key={reportEventsVersionRef.current} config={reportLayout} simulationData={reportData}></VisualComponent>}
+            </>
+        )
+    }
+}
+
+export class SimulationStartLayout extends View {
```

Comment:
Let's ban "Simulation" and just use "Sim".

Reply:
How about using 'report' universally. I already refer to the concept of a 'manually started simulation' as a 'report group' within the schema and code elsewhere. Would the name ReportGroupStatusLayout be a good replacement here or do you prefer 'sim'?

Reply:
There are two concepts: reports (subclass animations) and computes. When you press "Simulation start" you are staring a compute, not a report. This language needs to be made clear, since a compute maps to many reports. I don't like ReportGroup, because we already try to distinguish this on the server with compute and report and ReportGroup is not necessarily the only way computes are accessed. Sometimes you want to see if a compute is running. I think Jspec looks at a group of Elegant computes for twiss parameters to be used as input to Jspec.